src/config/src/meta/stream.rs

```diff
@@ -554,6 +562,44 @@ impl PartialEq for DistinctField {
 }
 impl Eq for DistinctField {}
 
+#[derive(Debug, Clone, Default, Serialize, Deserialize, ToSchema, PartialEq)]
+pub struct TimeRange {
+    pub start: i64,
+    pub end: i64,
+}
+
+impl TimeRange {
+    pub fn new(start: i64, end: i64) -> Self {
+        Self { start, end }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.start == 0 && self.end == 0
+    }
+
+    pub fn contains(&self, other: &Self) -> bool {
+        self.start <= other.start && self.end >= other.end
+    }
+
+    pub fn intersects(&self, other: &Self) -> bool {
+        self.start <= other.end && self.end >= other.start
+    }
```

Comment:
if the two ranges same, it will return `true`, then the next `split` function will return empty, then the `retention` will push an empty range.